### DISCUSSION TITLE
Addon Ideas #7 (2.0 Importations)

### DISCUSSION THREAD

**Discussion by SpryzenR** [ScratchAddons/ScratchAddons#6439] [Ideas] [answered]
Prior to this, in my previous ideas post (#6385), I gave a small list of old Scratch 2.0 things that could be addons in 3.0. So I thought, wouldn't it be nice to have a  bunch more????

### Addon Name: 
2.0 Importations (or something)

### Why it would be useful: 
Good for Scratch 2.0 users, the style would be converted to be more like a 3.0 style. It would be as if Scratch 2.0 was still there. It would be a selectable collection of old 2.0 things if Scratch 3.0 users wanna check out (3.0 polished) parts of Scratch 2.0. 

### List of (As of 7/24/23) Currently Available 2.0 Importation Addons: 

-Two column category menu by TheColaber

-Collapsing sprite properties by lisa_wolfgang (one idea of mine is to make it take up all the space in the sprites section like it does in 2.0 to show the space.)

-Display stage on the left side by NitroCipher/ZenithRougue

-Reverse order of the project controls 

### My List of Possible New 2.0 Importations:

-One column vector/bitmap tools

-Display vector/bitmap tools on the right side  

-Display old vector/bitmap icons (maybe give the option to apply the icons separately) 

-Add back the color presets (Maybe place them in the bottom of the editor or in the weird space I stated in my previous post #6385) 

-Add back the clear costume button

-Add back 2.0 backgrounds and sprites (ex: atom playground and old glasses sprite)

-Add back 2.0 text fonts (including separately the old marker font)

-Add back the sliders for the eraser, brush, line, and outline thickness.  

-Add back the Stamp button (for easier copy and paste)

-Organize the sound effects back into a drop down menu

-Move the Share button back into the file drop down menu

-Show Code Area Zoom Percentage below the Start Zoom Control Button. There are 2 settings for this. The first is to have it show forever or have it show for (amount of seconds; the minimum being 1 second and the max being 10 seconds) after you change the zoom. 

-An Updated 3.0 Tips Window that replaces it's newer Projects Tutorial Section counterpart that moves all of the updated tutorials to its window and covers all 3.0 blocks including the extensions blocks. (All of which is given examples of usages of the block) If this importation is turned on it will turn on the Hide Tutorials button addon by lisa_wolfgang and add a block help button to find the information on the block easier.

-Show Scratch version. (2.0 showed it below the fullscreen button but you could honestly put it anywhere that has space; you also could add the setting to show ScratchAddons current version as well.) 

Make this: 
![Screenshot (53)](https://github.com/ScratchAddons/ScratchAddons/assets/99098977/4c576d32-6186-4c1a-b536-26317f2bbc50)

Look like this: 
![Screenshot (55)](https://github.com/ScratchAddons/ScratchAddons/assets/99098977/3b8d89bf-9788-4337-a13e-dd36fef4690e)

Notice the selection and rotation visual difference (color, small circles instead of squares, and arrows for rotation instead of the circle) You could make an addon that could have these settings: 

-3.0 Selection --> 2.0 Selection. 
-Color for Selection (Hex and Slider). 
-Put arrow rotation above object (but if the 3.0 Selection --> 2.0 Selection is on then it will change to "put circle rotation below object")

Pt.2 (no 2.0 importations): #6441

**Comment by Samq64** [comment, marked as answer]
> One column vector/bitmap tools

This was recently added in #4018.

> Display old vector/bitmap icons (maybe give the option to apply the icons separately)

An addon that does that would probably also change the other editor icons.

> Add back the color presets

Do you mean https://github.com/ScratchAddons/ScratchAddons/issues/1807#issuecomment-796371324?

> Color for Selection (Hex and Slider)

We already have a hex colour input addon. I'm not exactly sure what you mean by slider.

**Reply by SpryzenR** [reply]
When is #4018 coming out it look cool

**Reply by WorldLanguages** [reply]
> This was recently added in https://github.com/ScratchAddons/ScratchAddons/pull/4018.

Huh, I wasn't really aware. Nice

**Reply by Samq64** [reply]
In 1.34. Hopefully during the first week of August.

**Reply by SpryzenR** [reply]
YESSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

**Comment by SpryzenR** [comment]
> > One column vector/bitmap tools
> 
> This was recently added in #4018.
> 
Ok I'm late to the party X_X
> > Display old vector/bitmap icons (maybe give the option to apply the icons separately)
> 

> An addon that does that would probably also change the other editor icons.
> 
Yeah
> > Add back the color presets
> 

> Do you mean [#1807 (comment)](https://github.com/ScratchAddons/ScratchAddons/issues/1807#issuecomment-796371324)?
> 
Pretty much yes 
> > Color for Selection (Hex and Slider)
> 
> We already have a hex colour input addon. I'm not exactly sure what you mean by slider.

Idk why I put slider in there but when you select in 3.0 there's a blue indicator that makes the selected object stand out so an idea is that maybe we could change the color of that.

**Comment by Secret-chest** [comment]
Also needed is always visible colour picker, and you should add coloured context menus, editor dark mode and custom block colours to the list of existing ones
